docs/design/coreclr/botr/readytorun-format.md

```diff
@@ -4,21 +4,37 @@ ReadyToRun File Format
 Revisions:
 * 1.1 - [Jan Kotas](https://github.com/jkotas) - 2015
 * 3.1 - [Tomas Rylek](https://github.com/trylek) - 2019
+* 4.1 - [Tomas Rylek](https://github.com/trylek) - 2020
 
 # Introduction
 
-This document describes ReadyToRun format implemented in CoreCLR as of June 2019.
+This document describes ReadyToRun format 3.1 implemented in CoreCLR as of June 2019 and not yet
+implemented proposed extensions 4.1 for the support of composite R2R file format. 
+**Composite R2R file format** has basically the same structure as the traditional R2R file format
+defined in earlier revisions except that the output file represents a larger number of input MSIL
+assemblies compiled together as a logical unit.
 
 # PE Headers and CLI Headers
 
-ReadyToRun images conform to CLI file format as described in ECMA-335, with the following
-customizations:
+**Single-file ReadyToRun images** conform to CLI file format as described in ECMA-335
+with the following customizations:
 
 - The PE file is always platform specific
 - CLI Header Flags field has set `COMIMAGE_FLAGS_IL_LIBRARY` (0x00000004) bit set
 - CLI Header `ManagedNativeHeader` points to READYTORUN_HEADER
 
-The image contains full copy of the IL and metadata that it was generated from.
+The COR header and ECMA 335 metadata pointed to by the COM descriptor data directory item
+in the COFF header represent a full copy of the input IL and MSIL metadata it was generated from.
+
+**Composite R2R files** currently conform to Windows PE executable file format as the
+native envelope. Moving forward we plan to gradually add support for platform-native
+executable formats (ELF on Linux, MachO on OSX) as the native envelopes. As a natural corollary
+There is no global CLI / COR header in the file. The ReadyToRun header structure is pointed to
```

Comment:
`there` or the previous `As a natural corollary` is perhaps a truncated sentence.

Reply:
Right, "there" was supposed to be part of the same sentence. Thanks for spotting, fixed in 3rd commit.